access-common/src/main/java/edu/unc/lib/dl/ui/service/LorisContentService.java

```diff
@@ -141,11 +156,38 @@ public void streamJP2(String simplepid, String region, String size, String rotat
         }
     }
 
+    public String getManifest(String id, String datastream, HttpServletRequest request) throws JsonProcessingException {
+        String[] url = request.getRequestURL().toString().split("\\/");
+        String urlBase = "https://" + url[2];
+        String manifestBase = urlBase + "/" + id;
+        ObjectMapper iiifMapper = new IiifObjectMapper();
+
+        Manifest manifest = new Manifest(manifestBase + "/manifest");
+
+        Canvas canvas = new Canvas(manifestBase);
+        String path = urlBase + "/jp2Proxy/" +
```

Comment:
It looks like this is reconstructing `request.getRequestURL()` with `/manifest` removed from the end. It seems like you could probably skip passing `datastream` into this method (and possibly `id` as well), and glue back together the pieces in `String[] url`. 
One other suggestion, we have `URIUtil.join` which can put together path segments in a URI, and will handle making sure the /'s are in the right places.